#!/usr/bin/env python

from math import pi as PI
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, "Wether to apply corrections or not",  True)
gen.add("obstacle_value", int_t, 0, "Value at which obstacles are represented in the recevied map", 100, 0, 255)
gen.add("inflation_distance", double_t, 0, "Distance arround map points where scan match points will be used to correct the pose", 0.15, 0.0, 0.5)
gen.add("localization_distance", double_t, 0, "Distance arround robot to be used in the localization scoring", 10.0, 0.1, 25.0)
gen.add("update_min_d", double_t, 0, "Euclidean distance before performing a pose registration", 0.1, 0.0, 5);  # name taken from AMCL, to maintain cohesion
gen.add("update_min_a", double_t, 0, "Angular distance before performing a pose registration", 0.1, 0.0, 2*PI); # name taken from AMCL, to maintain cohesion

exit(gen.generate("localization_checker", "localization_checker_node", "LocalizationChecker"))
